cardinal/events/events.go

```diff
@@ -9,10 +10,84 @@ import (
 	"github.com/go-openapi/runtime/middleware"
 	"github.com/gorilla/websocket"
 	"github.com/rs/zerolog/log"
+	ecslog "pkg.world.dev/world-engine/cardinal/ecs/log"
 )
 
-func CreateEventHub() *EventHub {
-	res := EventHub{
+type EventHub interface {
+	EmitEvent(event *Event)
+	FlushEvents()
+	ShutdownEventHub()
+	Run()
+}
+
+type loggingEventHub struct {
+	logger     *ecslog.Logger
+	eventQueue []*Event
+	running    atomic.Bool
+	broadcast  chan *Event
+	shutdown   chan bool
+	flush      chan bool
+}
+
+func (eh *loggingEventHub) EmitEvent(event *Event) {
+	eh.broadcast <- event
+}
+
+func (eh *loggingEventHub) FlushEvents() {
+	eh.flush <- true
+}
+
+func (eh *loggingEventHub) Run() {
+	if eh.running.Load() {
+		return
+	}
+	eh.running.Store(true)
+	for eh.running.Load() {
+		select {
+		case event := <-eh.broadcast:
+			eh.eventQueue = append(eh.eventQueue, event)
+		case <-eh.flush:
+			wg := sync.WaitGroup{}
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				acc := 0
+				for _, event := range eh.eventQueue {
+					eh.logger.Info().Msg(event.Message)
+					acc += 1
+				}
+				fmt.Printf("called %d times\n", acc)
```

Comment:
i think we can remove the acc and the print?

also, for the logged message, lets add a prefix that signals this is an event:

`eh.logger.Info().Msg("EVENT: " + event.Message)`